src/components/QueryEditor/QueryHeader.test.tsx

```diff
@@ -96,6 +97,14 @@ describe('QueryEditor', () => {
       expect(onChange).toHaveBeenCalledWith(expect.objectContaining({ rawMode: false }));
       expect(setDirty).toHaveBeenCalledWith(false);
     });
+
+    it('runs a query', async () => {
+      const onRunQuery = jest.fn();
+      render(<QueryHeader {...defaultProps} schema={schema} onRunQuery={onRunQuery} />);
+      await waitFor(() => screen.getByText('foo'));
+      screen.getByRole('button').click();
```

Comment:
I know there is just one button - for now - but maybe we can add a name property to look for the `run` button?